### DISCUSSION TITLE
DIANN Library using different digestion rules and PTMs

### DISCUSSION THREAD

**Discussion by PCFRH** [vdemichev/DiaNN#1064] [Q&A]
Hello,

First, I would like to thank everyone working on DIANN because that's an amazing software and the overall precursors/protein identifications are really good.

I recently noticed in the literature a publication dealing wiht the chemical modification of Cysteins into Pseudo-lysins . As a consequence, this reaction is adding an extra cleavage site for Trypsin and it also adds some extra modifications on the cysteins.

It means that the Trypsin will then cut after K,R and C, and we will have some fix modifications on cysteins ( + C(2)H(5)N or +43.0422 Da )

Using the GUI, in the additional option panel, can we add these 2 info to generate a Library taking into account these 2 extra parameters ?

If we do have to the command line version of DIANN, could you please suggest me parameters that have to be changed ?

Please let me know if you need some extra info. 

Best ,

Romain

**Comment by vdemichev** [comment]
Hi Romain,

Yes, should be possible. Please use the --cut command to specify the cleavage specificity and the --fixed-mod to modify the cysteine with the appropriate mass, when generating the spectral library https://github.com/vdemichev/DiaNN?tab=readme-ov-file#command-line-reference.

If the cysteine modification occurs only on some cysteines, then can consider using --var-mod instead of --fixed-mod, still adding cysteine to the --cut command but also increasing the max missed cleavage number to 2.

With the command line version, please just copy the command the GUI prints at the top of the log.

Best,
Vadim
